### PR TITLE
fix: polish deployment ux

### PR DESCRIPTION
**Pull request by lgalabru**
Address:
https://github.com/hirosystems/clarinet/issues/1129
https://github.com/hirosystems/clarinet/issues/1235

### CODE REVIEW

File: components/clarinet-deployments/src/onchain/mod.rs [hunk 1]
```diff
@@ -404,12 +404,7 @@ pub fn apply_on_chain_deployment(
 
     for batch_spec in deployment.plan.batches.iter() {
         let epoch = match batch_spec.epoch {
-            Some(epoch) => {
-                if network != StacksNetwork::Devnet {
-                    println!("warning: 'epoch' specified for a deployment batch is ignored when applying a deployment plan. This field should only be specified for deployments plans used to launch a devnet with 'clarinet devnet start'.");
-                }
-                epoch
-            }
+            Some(epoch) => epoch,
```
Comment (hugocaillard) [inline components/clarinet-deployments/src/onchain/mod.rs:407]: What about
```
let epoch = batch_spec.epoch.unwrap_or(default_epoch);
```

File: components/clarinet-cli/src/frontend/cli.rs [hunk 2]
```diff
@@ -828,7 +834,7 @@ pub fn main() {
                         match cmd {
                             DeploymentEvent::Interrupted(message) => {
                                 println!(
-                                    "{} Error publishing transactions: {}",
+                                    "{} Error publishing transactions (x): {}",
```
Comment (hugocaillard) [inline components/clarinet-cli/src/frontend/cli.rs:837]: Why this change?
I don't picture how it will look like

File: components/clarinet-deployments/src/onchain/mod.rs [hunk 3]
```diff
@@ -840,68 +835,90 @@ pub fn apply_on_chain_deployment(
...
-                                tracker.status = TransactionStatus::Confirmed;
-                                let _ = deployment_event_tx
-                                    .send(DeploymentEvent::TransactionUpdate(tracker.clone()));
+            // Handle Stacks releated checks
+            if stacks_tip_height > last_stacks_chain_check_at_height {
+                for (_, tracker) in ongoing_batch.iter_mut() {
+                    let TransactionStatus::Broadcasted(brodcasting_status, _) = &tracker.status
+                    else {
+                        continue;
+                    };
```
Comment (hugocaillard) [inline components/clarinet-deployments/src/onchain/mod.rs:858]: Is this a normal syntax?

File: components/clarinet-deployments/src/onchain/mod.rs [hunk 4]
```diff
@@ -219,7 +219,7 @@ pub enum TransactionCheck {
     NonceCheck(StandardPrincipalData, u64),
     ContractPublish(StandardPrincipalData, ContractName),
     // TODO(lgalabru): Handle Bitcoin checks
```
Comment (hugocaillard) [inline components/clarinet-deployments/src/onchain/mod.rs:221]: Should this comment be removed?

**Review by hugocaillard — COMMENTED** [review]
A couple of remarks and questions

File: components/clarinet-deployments/src/onchain/mod.rs (hunk 1 above)
Comment (lgalabru) [reply components/clarinet-deployments/src/onchain/mod.rs]: Addressed with latest

File: components/clarinet-cli/src/frontend/cli.rs (hunk 2 above)
Comment (lgalabru) [reply components/clarinet-cli/src/frontend/cli.rs]: Addressed with latest

File: components/clarinet-deployments/src/onchain/mod.rs (hunk 3 above)
Comment (lgalabru) [reply components/clarinet-deployments/src/onchain/mod.rs:854]: Yep, we don't want to execute the subsequent line if the chain height is still the same

File: components/clarinet-deployments/src/onchain/mod.rs (hunk 4 above)
Comment (lgalabru) [reply components/clarinet-deployments/src/onchain/mod.rs]: Addressed with latest

**Review by hugocaillard — APPROVED** [review]
LGTM
Thanks Ludo
